benchto-driver/src/main/java/com/teradata/benchto/driver/graphite/GraphiteClient.java

```diff
@@ -108,8 +108,18 @@ private URI buildLoadMetricsURI(Map<String, String> metrics, ZonedDateTime from,
         double[] dataPoints = new double[inputDataPoints.length];
         for (int i = 0; i < inputDataPoints.length; i++) {
             Double value = inputDataPoints[i][DATA_POINT_VALUE_INDEX];
-            // rarely graphite does not return value for given timestamp, use 0 in this case
-            dataPoints[i] = value == null ? 0 : value;
+            if (value == null) {
+                /*
+                 * Graphite returns null for an aggregation if *all* ingredients are null. We should query Graphite after
+                 * delay, so that all ingredients *are* present, so aggregation should not be null too.
+                 *
+                 * Note however, that non-null value for e.g. an average or sum does not mean that the value can be trusted.
+                 * Some data points can still be missing.
+                 */
+                LOGGER.error("null data point returned from Graphite");
```

Comment:
> Maybe there a way to obtain timestamp (though additional REST query) of latest data point for a given metrics and do validation then.

It's certainly doable, but iff we asked Graphite about single metric, no aggregation. But we don't do that. Calling side currently knows nothing about aggregates and it make it quite much more complex if it did.

Unfortunately Graphite doesn't seem to offer (I searched for this) an option to return null for an aggregate if any ingredient is null.